src/app/static/src/app/store/root/mutations.ts

```diff
@@ -38,7 +39,10 @@ export const mutations: MutationTree<RootState> = {
             hintrVersion: state.hintrVersion,
             language: state.language,
             adr: state.adr,
-            genericChart: state.genericChart,
+            genericChart: {
```

Comment:
For both Reset, and SetProject, we retain the previous state's genericChart metadata (which is only fetched once on app load) but clear the datasets - we could consider splitting the state into two modules to avoid treating different parts of module's state differently.  
We'll conditionally reset data if maxValid step < 2, as we know this is where generic chart is placed.